packages/buidler-core/src/internal/buidler-evm/provider/node.ts

```diff
@@ -1234,6 +1266,10 @@ export class BuidlerNode extends EventEmitter {
     return latestBlockTimestamp.eq(blockTimestamp);
   }
 
+  private async _setBlockTimestamp(block: Block, timestamp: BN) {
```

Comment:
Also, this may not be necessary. As it's already set in the block template, right?

Reply:
It is still needed as `mineEmptyBlock(timestamp)` needs to set the block's timestamp (overriding the `nextBlockTimestamp`) if timestamp param is not 0.